File: ruby/darts/darts.rb
class Darts
  def initialize(x, y)
    @radius = Math.sqrt(x**2 + y**2)
  end

  private

  attr_reader :radius

  public

  def score
    if radius <= 1 then 10
    elsif radius <= 5 then 5
    elsif radius <= 10 then 1
    else 0
    end
  end
end

File: javascript/darts/darts.js
export const score = (x, y) => {
  const radius = Math.sqrt(x**2 + y**2);

  if (radius <= 1) return 10;
  else if (radius <= 5) return 5;
  else if (radius <= 10) return 1;
  else return 0;
};
